src/components/Onboarding/Screens/Apps/TokensSettingsNative.js

```diff
@@ -0,0 +1,450 @@
+  const fieldsLayout = useFieldsLayout()
+
+  const { onBack, onNext, steps, step, onConfigChange } = useOnboardingState()
+
+  const [formError, setFormError] = useState()
+
+  const [tokenName, setTokenName] = useState('')
+  const [tokenSymbol, setTokenSymbol] = useState('')
+  const [commonPool, setCommonPool] = useState(0)
+  const [accountStake] = useState(0)
+
+  const [members, setMembers] = useState([['', accountStake]])
+
+  const handleTokenNameChange = useCallback(event => {
+    setFormError(null)
+    setTokenName(event.target.value)
+  }, [])
+
+  const handleTokenSymbolChange = useCallback(event => {
+    setFormError(null)
+    setTokenSymbol(event.target.value.trim().toUpperCase())
+  }, [])
+
+  const handleCommonPoolChange = useCallback(event => {
+    setFormError(null)
+    setCommonPool(event.target.value)
+  }, [])
+
+  const membersRef = useRef()
+  const [focusLastMemberNext, setFocusLastMemberNext] = useState(false)
+
+  useEffect(() => {
+    if (!focusLastMemberNext || !membersRef.current) {
+      return
+    }
+
+    setFocusLastMemberNext(false)
+
+    // This could be managed in individual MemberField components, but using
+    // the container with a .member class makes it simpler to manage, since we
+    // want to focus in three cases:
+    //   - A new field is being added.
+    //   - A field is being removed.
+    //   - The first field is being emptied.
+    //
+    const elts = membersRef.current.querySelectorAll('.member')
+    if (elts.length > 0) {
+      elts[elts.length - 1].querySelector('input').focus()
+    }
+  }, [focusLastMemberNext])
+
+  const focusLastMember = useCallback(() => {
+    setFocusLastMemberNext(true)
+  }, [])
+
+  const addMember = useCallback(() => {
+    setFormError(null)
+    setMembers(members => [...members, ['', accountStake]])
+    focusLastMember()
+  }, [accountStake, focusLastMember])
+
+  const removeMember = useCallback(
+    index => {
+      setFormError(null)
+      setMembers(members =>
+        members.length < 2
+          ? // When the remove button of the last field
+            // gets clicked, we only empty the field.
+            [['', accountStake]]
+          : members.filter((_, i) => i !== index)
+      )
+      focusLastMember()
+    },
+    [accountStake, focusLastMember]
+  )
+
+  const updateMember = useCallback((index, updatedAccount, updatedStake) => {
+    setFormError(null)
+    setMembers(members =>
+      members.map((member, i) =>
+        i === index ? [updatedAccount, updatedStake] : member
+      )
+    )
+  }, [])
+
+  const handleNext = useCallback(
+    event => {
+      event.preventDefault()
+
+      const error = validationError(tokenName, tokenSymbol, members, commonPool)
+      setFormError(error)
+
+      if (!error) {
+        onConfigChange('tokens', {
+          name: tokenName,
+          symbol: tokenSymbol,
+          holders: members.filter(
+            ([account, stake]) => isAddress(account) && stake > 0
+          ),
```

Comment:
Is the filtering necessary? Since we already ensure in `validationError` that all members follow the conditions?

Reply:
I deleted it